ext/php7/php7/engine_hooks.c

```diff
@@ -507,6 +507,14 @@ static ddtrace_span_fci *dd_fcall_begin_tracing_hook(zend_execute_data *call, dd
     span_fci->dispatch = dispatch;
     ddtrace_open_span(span_fci);
 
+    // default service to env var DD_SERVICE
+    zend_string *service = get_DD_SERVICE();
+    if (service && ZSTR_LEN(service)) {
+        zval *prop_service = ddtrace_spandata_property_service(&span_fci->span);
+        // todo: do we need to copy this?
+        ZVAL_STR_COPY(prop_service, service);
```

Comment:
If I followed the macro expansion correctly, then the `get_DD_SERVICE()` doesn't add a ref, so we need to. CI also passed. Seem right to you too, @bwoebi?

Reply:
yes, that's right. Most usages do not need a copy, hence we don't addref by default.